torch_xla/experimental/xla_sharding.py

```diff
@@ -319,8 +319,28 @@ def _get_sharding_type(partition_spec: Tuple[Union[int, None]],
   return sharding_type
 
 
-def _get_tile_assignment(mesh: Mesh) -> List[int]:
-  return mesh.get_logical_mesh().tolist()
+def _get_tile_assignment(mesh: Mesh,
+                         partition_spec: Tuple[Union[int, None]]) -> List[int]:
+  # Use Torch.tensor here to make use of the torch.transpose_
+  mesh_list_tensor = torch.tensor(mesh.get_logical_mesh().tolist())
+  # This is 2d tensor 3d mesh case, tile_assigniment will be ignore in favor of
+  # group_assignment and replication_groups.
```

Comment:
Does `group_assignment` already handle the reordering?

Reply:
nope lol I plan to support that in next pr.